http/http_client.hpp

```diff
@@ -626,10 +628,10 @@ class ConnectionInfo : public std::enable_shared_from_this<ConnectionInfo>
 
     void initializeConnection(bool ssl)
     {
-        conn = std::make_unique<boost::asio::ip::tcp::socket>(ioc);
+        conn = boost::asio::ip::tcp::socket(ioc);
         if (ssl)
         {
-            /* std::optional<boost::asio::ssl::context> sslCtx =
+            std::optional<boost::asio::ssl::context> sslCtx =
```

Comment:
this will have problem in downstream.  I see that downstream was not using proper ssl context as upstream for some reason. It is better to keep it that way

Reply:
Yes. This was commented out in downstream - to work with the HMC's SSL context.

Reply:
Why couldn't we have gone upstream with https://github.com/ibm-openbmc/bmcweb/commit/d104e6f83a28a291c4637f8d4bafa796c175b493  ? 

Let's just comment out for now. 
Then to me, this looks good.